sdk/communication/communication-phone-numbers/test/lro.update.aad.spec.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { Recorder, env, isPlaybackMode } from "@azure/test-utils-recorder";
+import { assert } from "chai";
+import { Context } from "mocha";
+import { PhoneNumberCapabilitiesRequest } from "../src";
+import { PhoneNumbersClient } from "../src/phoneNumbersClient";
+import { buildCapabilityUpdate } from "./utils";
+import {
+  canCreateRecordedClientWithToken,
+  createRecordedClientWithToken
+} from "./utils/recordedClient";
+
+describe("PhoneNumbersClient - lro - update [AAD]", function() {
+  const purchasedPhoneNumber = isPlaybackMode() ? "+14155550100" : env.AZURE_PHONE_NUMBER;
+  let recorder: Recorder;
+  let client: PhoneNumbersClient;
+
+  before(function(this: Context) {
+    if (!canCreateRecordedClientWithToken()) {
```

Comment:
this is more of an @DominikMe question. should we instead move tests that shouldn't run in browser into a separate node package?

Reply:
With the try/catch around `DefaultAzureCredential` we're avoiding the hard-code that DefaultAzureCredential only works in nodeJs. If it ever starts being supported in browser too, the way it's written now will just work. If we moved these tests into /node then we would need to make an effort to move the tests back if DefaultAzureCredential's node dependency goes away.

It kinda depends on whether we think `DefaultAzureCredential` not working in browser environments is an implementation detail that can change, or whether it's part of the public contract of the credential.

Reply:
I decided to take a look and see what other folks are doing!

app-configuration/test/public/auth.spec.ts does a pattern where if they can't retrieve the default token (via helper) it skips

Storage-blob has it's token test code for client instantation under node only (although it doesn't use DefaultAzureCredential, they create their own mock token).

In ai-form-regonizer/test/custom.spec.ts they default to an AzureKeyCredential if they can't create a DefaultAzureCredential. So this case they don't explicitly test instantiating using TokenCredential.

container-registry sidesteps this by using ClientSecretCredential instead of DefaultAzureCredential

DefaultAzureCredential's own code throws an error

const BrowserNotSupportedError = new Error(
  "DefaultAzureCredential is not supported in the browser. Use InteractiveBrowserCredential instead."
);